tests/queryset_pickle/tests.py

```diff
@@ -153,3 +153,23 @@ def test_unsupported_unpickle(self):
         msg = "Pickled queryset instance's Django version 1.0 does not match the current version %s." % get_version()
         with self.assertRaisesMessage(RuntimeWarning, msg):
             pickle.loads(pickle.dumps(qs))
+
+    def test_queryset_evaluation(self):
+        """
+        Pickling a QuerySet.query with an __in=inner_qs lookup,
+        shouldn't evaluate inner_qs.
+        """
+        for i in range(1, 5):
+            Group(name='Group {}'.format(i)).save()
+
+        Event(title='Event 1', group=Group.objects.all()[0]).save()
+
+        groups = Group.objects.all()
+        events = Event.objects.filter(group__in=groups)
```

Comment:
It doesn't work. Query doesn't hold information about queried fields thus it results query that queries all fields. Evaluation of your example query will result `OperationalError: only a single result allowed for a SELECT that is part of an expression`